scripts/build-macos.sh

```diff
@@ -1,41 +1,43 @@
 #!/bin/bash
 
-export GOOS=darwin 
+export GOOS=darwin
 export GOARCH=amd64
 
+PROJECT_ROOT="$(dirname "$(dirname "$(readlink -f "$0")")")"
+cd "${PROJECT_ROOT}" || exit 1
+
 echo "building swapd..."
-cd cmd/daemon 
-if ! go build -o swapd-amd64-darwin ; then
+cd cmd/daemon || exit 1
+if ! go build -o swapd-amd64-darwin; then
 	exit 1
 fi
 mv swapd ../..
 echo "done building swapd."
 
 echo "building swapcli..."
-cd ../client 
-if ! go build -o swapcli-amd64-darwin ; then 
+cd ../client || exit 1
+if ! go build -o swapcli-amd64-darwin; then
 	exit 1
 fi
 mv swapcli ../..
 echo "done building swapcli."
```

Comment:
I can simplify and consolidate a massive amount build code if we rename the folders in `cmd` to match the name of the executables they build. I didn't want to increase the diff in this PR though.

Reply:
feel free to do this in a follow up!